pkg/services/folder/folderimpl/folder.go

```diff
@@ -560,6 +567,18 @@ func (s *Service) validateParent(ctx context.Context, orgID int64, parentUID str
 		return folder.ErrMaximumDepthReached
 	}
 
+	// Create folder under itself is not allowed
+	if parentUID == UID {
```

Comment:
The `UID` field in `folder.CreateFolderCommand` is optional so we need to take into account also the case that it's empty.